lib/ansible/plugins/lookup/consul_kv.py

```diff
@@ -127,8 +127,8 @@ def run(self, terms, variables=None, **kwargs):
                 params = self.parse_params(term)
                 try:
                     url = os.environ['ANSIBLE_CONSUL_URL']
-                    validate_certs = os.environ['ANSIBLE_CONSUL_VALIDATE_CERTS'] or True
-                    client_cert = os.environ['ANSIBLE_CONSUL_CLIENT_CERT'] or None
+                    validate_certs = os.environ.get('ANSIBLE_CONSUL_VALIDATE_CERTS', True)
+                    client_cert = os.environ.get('ANSIBLE_CONSUL_CLIENT_CERT', None)
```

Comment:
instead of manually assembling all the options, use set_options/get_option  and all this will be resolved 'per option' in normal precedence order

Reply:
I just checked to code of the redis lookup, and yes that a better approach. I will update the pull request.